bookkeeper-server/src/main/java/org/apache/bookkeeper/replication/AuditorPlacementPolicyCheckTask.java

```diff
@@ -0,0 +1,297 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.bookkeeper.replication;
+
+import com.google.common.base.Stopwatch;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import org.apache.bookkeeper.client.BKException;
+import org.apache.bookkeeper.client.BookKeeperAdmin;
+import org.apache.bookkeeper.client.EnsemblePlacementPolicy;
+import org.apache.bookkeeper.client.api.LedgerMetadata;
+import org.apache.bookkeeper.conf.ServerConfiguration;
+import org.apache.bookkeeper.meta.LedgerManager;
+import org.apache.bookkeeper.meta.LedgerUnderreplicationManager;
+import org.apache.bookkeeper.meta.UnderreplicatedLedger;
+import org.apache.bookkeeper.net.BookieId;
+import org.apache.bookkeeper.proto.BookkeeperInternalCallbacks;
+import org.apache.bookkeeper.versioning.Versioned;
+import org.apache.zookeeper.AsyncCallback;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AuditorPlacementPolicyCheckTask extends AuditorTask {
+    private static final Logger LOG = LoggerFactory.getLogger(AuditorPlacementPolicyCheckTask.class);
+
+    private final long underreplicatedLedgerRecoveryGracePeriod;
+
+    AuditorPlacementPolicyCheckTask(ServerConfiguration conf,
+                                    AuditorStats auditorStats,
+                                    BookKeeperAdmin admin,
+                                    LedgerManager ledgerManager,
+                                    LedgerUnderreplicationManager ledgerUnderreplicationManager,
+                                    SubmitTaskHandler submitTaskHandler,
+                                    ShutdownTaskHandler shutdownTaskHandler) {
+        super(conf, auditorStats, admin, ledgerManager,
+                ledgerUnderreplicationManager, submitTaskHandler, shutdownTaskHandler);
+        this.underreplicatedLedgerRecoveryGracePeriod = conf.getUnderreplicatedLedgerRecoveryGracePeriod();
+    }
+
+    @Override
+    protected void runTask() {
+        try {
+            if (!isLedgerReplicationEnabled()) {
+                LOG.info("Ledger replication disabled, skipping placementPolicyCheck");
+                return;
+            }
+
+            Stopwatch stopwatch = Stopwatch.createStarted();
+            LOG.info("Starting PlacementPolicyCheck");
+            placementPolicyCheck();
+            long placementPolicyCheckDuration = stopwatch.stop().elapsed(TimeUnit.MILLISECONDS);
+            int numOfLedgersFoundNotAdheringInPlacementPolicyCheckValue =
+                    auditorStats.getNumOfLedgersFoundNotAdheringInPlacementPolicyCheck().get();
```

Comment:
Not use auditorStats.getNumOfLedgersFoundNotAdheringInPlacementPolicyCheck, we can define numOfLedgersFoundNotAdheringInPlacementPolicyCheck in AuditorPlacementPolicyCheckTask.
Some other fields also are, just define the guava field in AuditorStats

Reply:
I agree! Addressed done.